droid-hal-device.inc

```diff
@@ -829,20 +788,100 @@ fi
 %endif
 %endif
 
+%pre users
+#Compatibility with old droid-hal version - convert previous format
+#of droid-specific user/group information to current one.
+cd %{_libdir}/droid
+if [ $1 -gt 1 -a \
+     -e droid-user-remove.sh.installed -a \
+     -e droid-user-add.sh ]; then
+  #We've got the previous data in a shell-style generated text file. Convert.
+  #Only user/group name and ID is important, username=groupname, uid=gid.
+  sed -n -e 's/^groupadd -g \(.*\) \(.*\)/\2 \1/p' \
+      droid-user-add.sh > droid.ids.created || :
+  #Truncate the old file. rpm will remove it on its own.
+  echo '#!/bin/sh' > droid-user-remove.sh.installed
+fi
+
+%preun users
+# Only run this during final cleanup
```

Comment:
I wouldn't delete groups ever. Only if there is a group with differing group id from old one, then we could do

`groupmod --gid <new group id> <group name>`

Deleting groups was the problem to start with, trying to kludge with this is difficult. As the important thing is that whatever the used group array currently active has correct guids better to just update those if they change and leave possibly unused ones as is. This way for example the workaround below for system mode PulseAudio would be unneeded. As pulse was added at some point to audio group, audio group will continue existing and if the guid changes group members are intact.